process/block/preprocess/transactions.go

```diff
@@ -781,6 +778,34 @@ func (txs *transactions) AddTxsFromMiniBlocks(miniBlocks block.MiniBlockSlice) {
 	}
 }
 
+// AddTransactions adds the given transactions to the current block transactions
+func (txs *transactions) AddTransactions(txHandlers []data.TransactionHandler) {
+	for i, tx := range txHandlers {
+		senderShardID, err := txs.getShardFromAddress(tx.GetSndAddr())
+		if err != nil {
+			log.Warn("transactions.AddTransactions getShardFromAddress", "error", err.Error())
+			continue
+		}
+
+		receiverShardID, err := txs.getShardFromAddress(tx.GetRcvAddr())
+		if err != nil {
+			log.Warn("transactions.AddTransactions getShardFromAddress", "error", err.Error())
+			continue
+		}
+
+		txShardInfoToSet := &txShardInfo{senderShardID: senderShardID, receiverShardID: receiverShardID}
+		txHash, err := txs.marshalizer.Marshal(tx)
+		if err != nil {
+			log.Warn("transactions.AddTransactions Unmarshal", "error", err.Error())
+			continue
+		}
+
+		txs.txsForCurrBlock.mutTxsForBlock.Lock()
+		txs.txsForCurrBlock.txHashAndInfo[string(txHash)] = &txInfo{tx: txHandlers[i], txShardInfo: txShardInfoToSet}
```

Comment:
could have used the `tx` variable and removed the `i` usage since the slice is done on an interface, and thus, the tx will always be assigned to a new pointer value.